app/views/QRScan.js

```diff
@@ -0,0 +1,291 @@
+  let icon;
+  switch (status) {
+    case StateEnum.DEFAULT:
+      icon = null;
+      break;
+    case StateEnum.SCAN_SUCCESS:
+      icon = StateNoContact;
+      break;
+    case StateEnum.SCAN_FAIL:
+      icon = StateAtRisk;
+      break;
+    case StateEnum.SCAN_IN_PROGRESS:
+      icon = StateNoContact;
+      break;
+  }
+  return (
+    <SvgXml xml={icon} width={size ? size : 80} height={size ? size : 80} />
+  );
+};
+
+const height = Dimensions.get('window').height;
+
+class QRScan extends Component {
+  constructor(props) {
+    super(props);
+
+    if (isPlatformAndroid()) {
+      StatusBar.setBackgroundColor(Colors.TRANSPARENT);
+      StatusBar.setBarStyle('light-content');
+      StatusBar.setTranslucent(true);
+    }
+
+    this.state = {
+      currentState: StateEnum.DEFAULT,
+    };
+  }
+
+  componentDidMount() {
+    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+    const handleStateChange = () => {
+      if (this.props.navigation.isFocused()) {
+        this.onNavigate();
+      } else {
+        this.setState({ currentState: StateEnum.DEFAULT });
+      }
+    };
+    this.unsubscribeFocus = this.props.navigation.addListener(
+      'focus',
+      handleStateChange,
+    );
+    this.unsubscribeState = this.props.navigation.addListener(
+      'state',
+      handleStateChange,
+    );
+  }
+
+  componentWillUnmount() {
+    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
+    this.unsubscribeFocus();
+    this.unsubscribeState();
+  }
+
+  onNavigate() {
+    if (this.props.route && this.props.route.params) {
+      const { latitude, longitude } = this.props.route.params;
+      if (typeof latitude !== 'undefined' && typeof longitude !== 'undefined') {
+        this.saveCoordinates(latitude, longitude);
+      } else {
+        this.setState({ currentState: StateEnum.SCAN_IN_PROGRESS });
+      }
+    } else {
+      this.setState({ currentState: StateEnum.SCAN_IN_PROGRESS });
+    }
+  }
+
+  onScanSuccess(e) {
+    const url = e && e.data;
+    const split_1 = url && url.split('/qr/');
+    const split_2 = split_1 && split_1.length === 2 && split_1[1].split('/');
+    const latitude = split_2 && split_2.length === 2 && split_2[0];
+    const longitude = split_2 && split_2.length === 2 && split_2[1];
+    if (typeof latitude !== 'undefined' && typeof longitude !== 'undefined') {
+      this.saveCoordinates(latitude, longitude);
+    } else {
+      this.setState({ currentState: StateEnum.SCAN_FAIL });
+    }
+  }
+
+  saveCoordinates(latitude, longitude) {
+    if (this.isValidCoordinates(latitude, longitude)) {
+      LocationServices.saveLocation({
+        latitude: Number(latitude),
+        longitude: Number(longitude),
+        time: Date.now(),
+      });
+      this.setState({ currentState: StateEnum.SCAN_SUCCESS });
+    } else {
+      this.setState({ currentState: StateEnum.SCAN_FAIL });
+    }
+  }
+
+  isValidCoordinates(latitude, longitude) {
```

Comment:
feels like we can pull this out into a helper method. or maybe just dump it in `LocationService.js`

Reply:
Seems appropriate as a helper method, so I created a new `Location` helper.

Reply:
awesome 👍